system-tests/lib/cre/contracts/contracts.go

```diff
@@ -492,48 +495,72 @@ func DeployFeedsConsumer(testLogger zerolog.Logger, input *types.DeployFeedConsu
 		return nil, errors.Wrap(err, "input validation failed")
 	}
 
-	output, err := keystone_changeset.DeployFeedsConsumer(*input.CldEnv, &keystone_changeset.DeployFeedsConsumerRequest{
-		ChainSelector: input.ChainSelector,
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to deploy feeds_consumer contract")
+	deployConfig := df_changeset_types.DeployConfig{
+		ChainsToDeploy: []uint64{input.ChainSelector},
+		Labels:         []string{"data-feeds"},
+	}
+	dfOutput, dfErr := df_changeset.DeployCacheChangeset.Apply(*input.CldEnv, deployConfig)
+	if dfErr != nil {
+		return nil, errors.Wrap(dfErr, "failed to deploy data feed cache contract")
 	}
 
-	err = input.CldEnv.ExistingAddresses.Merge(output.AddressBook)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to merge address book")
+	mergeErr := input.CldEnv.ExistingAddresses.Merge(dfOutput.AddressBook)
```

Comment:
I'm not an expert on CLD @karen-stepanyan can you help me here? 🙏 
Is `Apply`, `Merge`, `AddressesForChain` and then looping over addresses to find `DataFeedsCache` the way we supposed to do it?

Reply:
You can skip merging and find the address from `dfOutput.AddressBook`. Still need to use AddressesForChain and do a loop. This is how current version of Addressbook works.

Reply:
we can't skip merging, because we pass that environment to other changeset, which expect DF Cache contract address to exist in the address book :/